java/java.hints/src/org/netbeans/modules/java/hints/errors/EnablePreview.java

```diff
@@ -0,0 +1,152 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.netbeans.modules.java.hints.errors;
+
+import com.sun.source.util.TreePath;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import javax.lang.model.SourceVersion;
+import org.netbeans.api.annotations.common.NonNull;
+import org.netbeans.api.java.platform.JavaPlatform;
+import org.netbeans.api.java.platform.JavaPlatformManager;
+import org.netbeans.api.java.queries.SourceLevelQuery;
+import org.netbeans.api.java.source.ClasspathInfo.PathKind;
+import org.netbeans.api.java.source.CompilationInfo;
+import org.netbeans.modules.java.hints.spi.ErrorRule;
+import org.netbeans.spi.editor.hints.ChangeInfo;
+import org.netbeans.spi.editor.hints.Fix;
+import org.openide.filesystems.FileObject;
+import org.openide.util.NbBundle;
+import org.openide.util.Parameters;
+import org.netbeans.modules.java.hints.spi.preview.PreviewEnabler;
+import org.netbeans.modules.java.hints.spi.preview.PreviewEnabler.Factory;
+import org.openide.modules.SpecificationVersion;
+import org.openide.util.Lookup;
+
+/**
+ * Handle error rule "compiler.err.preview.feature.disabled.plural" and provide
+ * the fix for Single Source Java File.
+ *
+ * @author Arunava Sinha
+ */
+public class EnablePreview implements ErrorRule<Void> {
+
+    private static final Set<String> ERROR_CODES = Collections.unmodifiableSet(new HashSet<String>(Arrays.asList(
+            "compiler.err.preview.feature.disabled",           //NOI18N
+            "compiler.err.preview.feature.disabled.plural"))); // NOI18N
```

Comment:
tested on JDK 20 and this doesn't trigger the hint atm:
```java
    Thread.Builder.OfVirtual ofVirtual = Thread.ofVirtual();
```
the compiler error:
```
COMPILATION ERROR : 
-------------------------------------------------------------
com/mycompany/mavenproject1/Mavenproject1.java:[15,15] java.lang.Thread.Builder is a preview API and is disabled by default.
  (use --enable-preview to enable preview APIs)
com/mycompany/mavenproject1/Mavenproject1.java:[15,23] java.lang.Thread.Builder.OfVirtual is a preview API and is disabled by default.
  (use --enable-preview to enable preview APIs)
com/mycompany/mavenproject1/Mavenproject1.java:[15,52] ofVirtual() is a preview API and is disabled by default.
  (use --enable-preview to enable preview APIs)
3 errors 
```
maybe there is a new error code now? Or does this only work for lang features?

Reply:
Uh, good catch! Added a new key.